portal-bridge/src/bridge/era1.rs

```diff
@@ -79,50 +85,76 @@ impl Era1Bridge {
+            BridgeMode::FourFours(FourFoursModeType::Epoch(epoch)) => {
+                self.launch_epoch(epoch).await
+            }
+            _ => panic!("4444s bridge only supports 4444s modes."),
+        }
+        info!("Bridge mode: {:?} complete.", self.mode);
+    }
+
+    pub async fn launch_random(&self) {
         for era1_path in self.era1_files.clone().into_iter() {
-            info!("Processing era1 file at path: {:?}", era1_path);
-            // We are using a semaphore to limit the amount of active gossip transfers to make sure
-            // we don't overwhelm the trin client
-            let gossip_send_semaphore = Arc::new(Semaphore::new(GOSSIP_LIMIT));
+            self.gossip_era1(era1_path).await;
+        }
+    }
 
-            let raw_era1 = self
-                .http_client
-                .get(era1_path.clone())
-                .recv_bytes()
-                .await
-                .unwrap_or_else(|_| panic!("unable to read era1 file at path: {era1_path:?}"));
-            let era1 = Era1::deserialize(&raw_era1).expect("to be able to deserialize era1 file");
-            let epoch_index = era1.block_tuples[0].header.header.number / EPOCH_SIZE as u64;
-            info!("Era1 file read successfully, gossiping block tuples for epoch: {epoch_index}");
-            let epoch_acc = match self.get_epoch_acc(epoch_index).await {
-                Ok(epoch_acc) => epoch_acc,
-                Err(e) => {
-                    error!("Failed to get epoch acc for epoch: {epoch_index}, error: {e}");
-                    continue;
-                }
-            };
-            let master_acc = Arc::new(self.header_oracle.master_acc.clone());
-            let mut serve_block_tuple_handles = vec![];
-            for block_tuple in era1.block_tuples {
-                let permit = gossip_send_semaphore
-                    .clone()
-                    .acquire_owned()
-                    .await
-                    .expect("to be able to acquire semaphore");
-                let serve_block_tuple_handle = Self::spawn_serve_block_tuple(
-                    self.portal_clients.clone(),
-                    block_tuple,
-                    epoch_acc.clone(),
-                    master_acc.clone(),
-                    permit,
-                );
-                serve_block_tuple_handles.push(serve_block_tuple_handle);
+    pub async fn launch_epoch(&self, epoch: u64) {
+        let epoch_string = epoch.to_string() + "-";
+        for era1_path in self.era1_files.clone().into_iter() {
+            if era1_path.contains(&epoch_string) {
+                self.gossip_era1(era1_path).await;
+                return;
+            }
+        }
+        panic!("4444s bridge couldn't find request epoch on era1 file server")
+    }
+
+    pub async fn gossip_era1(&self, era1_path: String) {
+        info!("Processing era1 file at path: {:?}", era1_path);
```

Comment:
I think this should be `info!("Processing era1 file at path: {era1_path:?}");` ... but also clippy should be catching this?